src/main/crc.ts

```diff
@@ -0,0 +1,54 @@
+// ==LICENSE-BEGIN==
+// Copyright 2017 European Digital Reading Lab. All rights reserved.
+// Licensed to the Readium Foundation under one or more contributor license agreements.
+// Use of this source code is governed by a BSD-style license
+// that can be found in the LICENSE file exposed on Github (readium) in the project repository.
+// ==LICENSE-END==
+
+import { crc32 } from "crc";
+import * as yauzl from "yauzl";
+
+type TCrcFile = [string, string];
```

Comment:
Just a thought: in my tests, the CRC numbers are quite short and human-readable. I would therefore prefer to define `type TCrcFile = [string, number]`, and I would postpone the use of `.toString(16)` to the last minute inside the `end` event handler (i.e. `crc32(curr[1].toString(16), prev)`), instead of the current code `fileArray.push([entry.fileName, (entry.crc32 || 0).toString(16)]);` inside the `entry` event handler.

This way, the `entry` event is super simple (collects raw data), and the `end` event contains more logic to process the collected data.

Reply:
fix 84970dd8